### PR TITLE
ft(cart.tsx):buyer add items to cart 187364824

### PR DESCRIPTION
**Pull request by tuyishimehope**
## ℹ️ Description

Please write about the changes implemented in this PR. Also write a short description about why the change is implemented.

[##### 📹 Loom video link: _URL here_](https://www.loom.com/share/de68e3dcb129433aa41f0f5c22120305)

## 🧪 How can this be tested?

Write a short description on how others can test the changes implemented in the PR

## 📸 Screenshots

image
image
image
image
image

## 📍 Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Aldot-02 — COMMENTED** [review]
Geat Work @tuyishimekyrie,

Consider adding the tests for your new feature, screenshoots, as well as a simple demo of the feature working live.

File: src/components/rootcomponents/ShopProduct.tsx [hunk 1]
```diff
@@ -1,19 +1,53 @@
...
+import { useToken } from '../../hooks/useToken';
 
 const ShopProduct: React.FC = () => {
+  const { token, user } = useToken();
+  const { data: products, error, isLoading } = useGetAllProductsQuery({});
+  const [addToCart, { isLoading: addtocartLoading }] = useAddToCartMutation();
+
+  const handleAddToCart = async (product: CartProduct) => {
+    if (!user || !token) {
+      toast.error('Please Log In');
```
Comment (shebz2023) [inline src/components/rootcomponents/ShopProduct.tsx:22]: setTimeout(() => {
          navigate('/auth/login') 
 }, 2500);

File: src/components/rootcomponents/ShopProduct.tsx [hunk 2]
```diff
@@ -1,19 +1,53 @@
...
+      return;
+    }
+    const response = await addToCart({
+      productid: product.product_id,
+      quantity: product.quantity,
+    });
+    console.log(response);
```
Comment (shebz2023) [inline src/components/rootcomponents/ShopProduct.tsx]: remove consoles

**Review by shebz2023 — COMMENTED** [review]
Great work @tuyishimekyrie ,some suggestions below

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by Aldot-02 — DISMISSED** [review]
LGTM

File: src/components/rootcomponents/ProductRow.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          <span className="py-1 px-4">
+            <button className="px-2 py-1  bg-slate-400 hover:bg-slate-600 text-white rounded">
+              -
+            </button>
+            <span className="mx-2 text-sm text-slate-500">
+              {product.quantity}
+            </span>
+            <button className="px-2 py-1 bg-slate-400 hover:bg-slate-600 text-white rounded">
+              +
+            </button>
```
Comment (SergeInGithub) [inline src/components/rootcomponents/ProductRow.tsx]: Why don't you use the button component we have?

File: src/pages/rootpages/Cart.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          <h1 className="px-4 py-2">Shopping Cart</h1>
+          <TableHeader />
+          <TableBody products={cartItems} />
+        </div>
+        <div className="bg-white p-8 rounded-md">
+          <h1 className="border-b border-slate-200">Order Summary</h1>
+          <div className="flex justify-between mt-6">
+            <p>Discount</p>
+            <p>$0.00</p>
+          </div>
```
Comment (SergeInGithub) [inline src/pages/rootpages/Cart.tsx]: This is a component. Stop repeating yourself

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Please work on the reviews provided by me and your teammates @tuyishimekyrie

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Please work on the changes requested by your colleagues! @tuyishimekyrie

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**